### PR TITLE
Bug fix/cleanup 20201003

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

* Slightly improve parallelism for creating agency callbacks, by not creating the pseudorandom callback ids under the lock.
* Avoid unnecessary comparisons in AgencyCallback::checkValue(...).
* Added metric `arangodb_agency_callback_registered counter` for tracking the total number of agency callbacks that were registered.
* BE -> AE

- [ ] :hankey: Bugfix 
- [ ] :pizza: New feature 
- [x] :hammer: Refactoring 
- [x] :book: CHANGELOG entry made
- [x] :muscle: The behavior in this PR was *manually tested*
- [ ] :computer: The behavior change can be verified via automatic tests

#### Backports:

- [x] No backports required

### Testing & Verification

- [x] This change is a trivial rework / code cleanup without any test coverage.

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12089/

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,6 +1,9 @@
 devel
 -----
 
+* Added metrics `arangodb_agency_callback_registered counter` for tracking the
```
Comment (jsteemann) [inline CHANGELOG:4]:
```suggestion
* Added metric `arangodb_agency_callback_registered counter` for tracking the
```

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12109/

**Comment by jsteemann** [comment]
Tests blue

**Review by kvahed — APPROVED** [review]
LGTM

**Pull request merged**